lib/lists.nix

```diff
@@ -687,6 +687,32 @@ rec {
    */
   unique = foldl' (acc: e: if elem e acc then acc else acc ++ [ e ]) [];
 
+
+  /* Check if list contains only unique elements. O(n) complexity.
```

Comment:
> attrset merge alone is O(n)

i did not expect that. Thought it would be O(1).

But then i agree, that this is not really useful.
Would we be better of just calculating the unique list and comparing the length of the calculated and the original one?

Reply:
Sorry but I do not really get your proposal.

Reply:
To be honest I am not sure if I full understand your code.

It seems the right direction, but what does `{ r = true; last = (dummy: 0); }` this part mean? is this some initialization before the fold?